JAVA_HOME did not get updated

If I switch between java versions, `JAVA_HOME` did not get updated.

```
❯ asdf current
java           adopt-openjdk-13.0.2+8 (set by /Users/jivanov/.tool-versions)

❯ java -version
openjdk version "13.0.2" 2020-01-14
OpenJDK Runtime Environment AdoptOpenJDK (build 13.0.2+8)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 13.0.2+8, mixed mode, sharing)

❯ echo $JAVA_HOME
/Users/jivanov/.asdf/installs/java/adopt-openjdk-13.0.2+8

❯ asdf global java adopt-openjdk-8u242-b08

❯ asdf current
java           adopt-openjdk-8u242-b08 (set by /Users/jivanov/.tool-versions)

❯ java -version
openjdk version "1.8.0_242"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_242-b08)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.242-b08, mixed mode)

❯ echo $JAVA_HOME
/Users/jivanov/.asdf/installs/java/adopt-openjdk-13.0.2+8
```

@halcyon How about my solution? It worked with both `system` java and `asdf` provider java
```
asdf_update_java_home() {
  # shellcheck disable=SC2046
  JAVA_HOME=$(realpath $(dirname $(readlink -f $(asdf which java)))/../)
  export JAVA_HOME
}
autoload -U add-zsh-hook
add-zsh-hook precmd asdf_update_java_home
```
https://github.com/thuandt/zsh-config/blob/master/functions.sh#L3-L7
https://github.com/thuandt/zsh-config/blob/master/my-zsh-config.plugin.zsh#L103

--- a/README.md
+++ b/README.md
@@ -36,8 +36,12 @@ Select an installed candidate for use like this:
 ## JAVA_HOME
 To set JAVA_HOME in your shell's initialization add the following:
 
-`. ~/.asdf/plugins/java/set-java-home.sh`
+`. ~/.asdf/plugins/java/set-java-home.bash`
 
-for fish shell, instead use:
+For zsh shell, instead use:
+
+`. ~/.asdf/plugins/java/set-java-home.zsh`
+
+For fish shell, instead use:
 
 `. ~/.asdf/plugins/java/set-java-home.fish`

--- a/set-java-home.bash
+++ b/set-java-home.bash
@@ -1,0 +1,14 @@
+asdf_update_java_home() {
+  local java_path
+  java_path="$(asdf which java)"
+  if [[ -n "${java_path}" ]]; then
+    export JAVA_HOME
+   JAVA_HOME="$(dirname "$(dirname "$(realpath "${java_path}")")")"
+  fi
+}
+
+prompt_command() {
+  asdf_update_java_home
+}
+
+export PROMPT_COMMAND="${PROMPT_COMMAND}; prompt_command"

--- a/set-java-home.fish
+++ b/set-java-home.fish
@@ -1,4 +1,8 @@
-asdf current java 2>&1 > /dev/null
-if test $status -eq 0
-    set -x JAVA_HOME (asdf where java)
+function asdf_update_java_home --on-event fish_prompt
+  set --local java_path (asdf which java)
+  if test -n "$java_path"
+    set --local full_path (realpath "$java_path")
+    set --local full_path_dir (dirname "$full_path")
+    set --export JAVA_HOME (dirname "$full_path_dir")
+  end
 end

--- a/set-java-home.sh
+++ b/set-java-home.sh
@@ -1,5 +1,0 @@
-asdf current java 2>&1 > /dev/null
-if [[ "$?" -eq 0 ]]
-then
-    export JAVA_HOME=$(asdf where java)
-fi

--- a/set-java-home.zsh
+++ b/set-java-home.zsh
@@ -1,0 +1,11 @@
+asdf_update_java_home() {
+  local java_path
+  java_path="$(asdf which java)"
+  if [[ -n "${java_path}" ]]; then
+    export JAVA_HOME
+    JAVA_HOME="$(dirname "$(dirname "$(realpath "${java_path}")")")"
+  fi
+}
+
+autoload -U add-zsh-hook
+add-zsh-hook precmd asdf_update_java_home

